File: intermediate-package/configuring-our-integration-tests/configuring-application-context/spring/src/integration-test/java/com/testwithspring/intermediate/message/GetMessageTest.java
package com.testwithspring.intermediate.message;

import com.testwithspring.intermediate.IntegrationTest;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = {MessageContext.class})
//@ContextConfiguration(locations = {"classpath:message-context.xml"})
@Category(IntegrationTest.class)
public class GetMessageTest {

    @Autowired
    private MessageService service;

    @Test
    public void shouldReturnHelloWorldMessage() {
        String message = service.getMessage();
        assertThat(message).isEqualTo("Hello World!");
    }
}

File: master-package/bonus-webinars-2019/running-kotlin-tests-with-maven/src/integration-test/kotlin/com/testwithspring/master/kotlin/GetMessageTest.kt
package com.testwithspring.master.kotlin

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test

@Tag("integrationTest")
class GetMessageTest {

    private lateinit var service: MessageService

    @BeforeEach
    fun configureSystemUnderTest() {
        service = MessageService()
    }

    @Test
    @DisplayName("Should return the correct message")
    fun shouldReturnCorrectMessage() {
        val message = service.getMessage()
        Assertions.assertThat(message).isEqualTo("Hello World!")
    }
}